File: scalechain-codec-proto/src/test/scala/io/scalechain/blockchain/proto/codec/messages/MempoolSpec.scala
package io.scalechain.blockchain.proto.codec.messages

import io.scalechain.blockchain.proto.Mempool
import io.scalechain.blockchain.proto.codec._
import io.scalechain.util.HexUtil._
import scodec.bits.BitVector

/**
    [Bitcoin Core Packets Not Captured]
 */
class MempoolSpec extends EnvelopeTestSuite[Mempool]  {

  val codec = MempoolCodec.codec

  val envelopeHeader = bytes("""
                             """)

  val payload = bytes("")

  val envelope = BitcoinMessageEnvelope(
    Magic.MAIN,
    "mempool",
    payload.length.toInt,
    Checksum.fromHex(""),
    BitVector.view(payload)
  )

  val message = null//Mempool()

}

File: scalechain-proto-codec/src/test/kotlin/io/scalechain/blockchain/proto/codec/messages/MempoolSpec.kt
package io.scalechain.blockchain.proto.codec.messages

import io.kotlintest.KTestJUnitRunner
import io.scalechain.blockchain.proto.*
import io.scalechain.blockchain.proto.codec.*
import io.scalechain.util.HexUtil.bytes
import org.junit.runner.RunWith


/**
 *  <Bitcoin Core Packets Not Captured>
 *  No payload.
 */

@RunWith(KTestJUnitRunner::class)
class MempoolSpec : PayloadTestSuite<Mempool>()  {

  override val codec = MempoolCodec

  override val payload = bytes("") // No payload.

  override val message = Mempool()

}
